torch/_inductor/codegen/triton.py

```diff
@@ -636,7 +636,8 @@ def to_dtype_bitcast(x, dtype: torch.dtype, src_dtype: torch.dtype):
         if src_dtype in (torch.float16, torch.bfloat16):
             triton_src_dtype = str(src_dtype).split(".")[-1]
             cast_x = f"{x}.to(tl.{triton_src_dtype})"
-            cast_x = f"{cast_x}.to({triton_dtype}, bitcast=True)"
+            if triton_dtype != "tl.float32":
+                cast_x = f"{cast_x}.to({triton_dtype}, bitcast=True)"
```

Comment:
I think this wrong for fp16.view(dtype=torch.bfloat16). Would have flushed out by parameterized testing.

Why are we adding this line?

Reply:
fp16.view(bfloat16) should be correct because `triton_compute_type` change the bfloat16 to float32. https://github.com/pytorch/pytorch/blob/7bda23ef8495a22bde8b7861f5d4f3f015cee483/torch/_inductor/codegen/triton.py#L1725-L1726

int16.view(bfloat16) is wrong because bfloat16 is converted to float32 then it was trying to bitcast int16->float32.
https://github.com/pytorch/pytorch/pull/128883/files#diff-ea3834aeda2e22f49dd5ba0becd456b449b10d17dfe64c34e89072701c82873dL642
It is fixed in https://github.com/pytorch/pytorch/commit/c638822f72e70db19ae20b044f09ccb6941255a5.